### PR TITLE
Fix #166 to use specified tokenizer when tagging.

### LINKED ISSUE #166: TextBlob object is not using explicitly specified tokenizer when tagging method is called
I was trying to use TextBlob object with explicitly passed WordPunctTokenizer from nltk. Tokens property is returned as expected. Yet when calling tags or pos_tags properties, the object falls back to using a simple WordTokenizer before tagging as specified in NLTKTagger code:
https://github.com/sloria/TextBlob/blob/dev/textblob/en/taggers.py#L33

### PR DESCRIPTION
**Pull request by jschnurr**
Let me summarise my approach;  welcome any feedback.

For reference - the [pos_tag](https://github.com/sloria/TextBlob/blob/dev/textblob/blob.py#L450) method on BaseBlob populates both the **pos_tag** and **tags** properties.

Currently, **pos_tag** provides the raw string to the **pos_tagger** defined on the object; it's neither tokenized nor broken down into sentences.  The tagger class (*PatternTagger* or default *NLTKTagger*) is fully responsible for deriving the part-of-speech tags using only string input.

Visualized as an NLP pipeline, it looks like this:
```raw
TextBlob -> .raw -> tagger -> tags
```

In this PR, we use the **.sentences** property to first break the object down, and then pass each sentence blob to the tagger. The tagger, now having a BaseBlob instead of just a string, can choose to use the tokenizer, depending on the implementation.

As a pipeline:
```raw
TextBlob -> Sentence -> tokenizer (optional) -> tagger -> tags
```

Based on their underlying dependencies, **NLTKTagger** will *use* the tokenizer, while **PatternTagger** will *ignore* it.

Here's what it looks like.

Before:
```python
>>> blob = tb.TextBlob("Good muffins cost $3.88\nin New York.", tokenizer=nltk.tokenize.regexp.WordPunctTokenizer())
>>> blob.words
WordList(['Good', 'muffins', 'cost', '3.88', 'in', 'New', 'York'])                                                                                                                                                
>>> blob.tokens                                                                                                                                                                                                   
WordList(['Good', 'muffins', 'cost', '$', '3', '.', '88', 'in', 'New', 'York', '.'])
>>> blob.tags
[('Good', u'JJ'), ('muffins', u'NNS'), ('cost', u'VBP'), ('3.88', u'CD'), ('in', u'IN'), ('New', u'NNP'), ('York', u'NNP')]
```

After:
```python
>>> blob = tb.TextBlob("Good muffins cost $3.88\nin New York.", tokenizer=nltk.tokenize.regexp.WordPunctTokenizer())
>>> blob.words
WordList(['Good', 'muffins', 'cost', '3.88', 'in', 'New', 'York'])
>>> blob.tokens
WordList(['Good', 'muffins', 'cost', '$', '3', '.', '88', 'in', 'New', 'York', '.'])
>>> blob.tags
[('Good', u'JJ'), ('muffins', u'NNS'), ('cost', u'VBP'), ('3', u'CD'), ('88', u'CD'), ('in', u'IN'), ('New', u'NNP'), ('York', u'NNP')]
```
Notice **$3.88** is handled differently by **blob.tags**, since the former uses the default tokenizer, and the latter honors the tokenizer set on the BaseBlob object.

### CODE REVIEW

File: textblob/en/taggers.py [hunk 1]
```diff
@@ -3,10 +3,11 @@
 from __future__ import absolute_import
 
 import nltk
+import six
```
Comment (sloria) [inline textblob/en/taggers.py]: Use `textblob.compat` instead of `six`.

**Comment by sloria** [comment]
Also, getting this branch up-to-date with master should fix the Travis build.

**Comment by jschnurr** [comment]
Ok @sloria, should be all good now.

**Comment by sloria** [comment]
Thanks! This is in 0.14.0.

By the way, I'm starting to use something along the lines of the [Pull Request Hack](http://felixge.de/2013/03/11/the-pull-request-hack.html). Basically, if you get a PR merged in, you get an invite to have commit access.

No pressure to do any more work than you already have. Just want to open the door to further collaboration. =)

**Pull request merged**